### PR TITLE
Back 1197 add label on register account

### PR DESCRIPTION
**Pull request by mbasire-ledger**
## What is this about?

Clients of the lama api are able to add a label using the register account api ( POST /accounts/ ).
The label is available in account info when calling 
 - GET /accounts/{uuid}
 - GET /accounts

## Cute picture of lama
lama

### CODE REVIEW

File: account-manager/src/main/resources/db/migration/V3__account_info_label.sql [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+        coin,
+        sync_frequency,
+        sync_id,
+        status,
+        payload,
+        updated,
+        label
+    FROM account_info JOIN account_sync_event USING (account_id)
+    ORDER BY account_id, updated DESC
+);
```
Comment (vbergeron-ledger) [inline account-manager/src/main/resources/db/migration/V3__account_info_label.sql:17]: Add newline

File: account-manager/src/main/scala/co/ledger/lama/manager/Service.scala [hunk 2]
```diff
@@ -188,7 +190,8 @@ class Service(val db: Transactor[IO], val coinConfigs: List[CoinConfig])
         UuidUtils.uuidToBytes(accountInfo.id),
         accountInfo.key,
         accountInfo.syncFrequency.toSeconds,
-        lastSyncEventProto
+        lastSyncEventProto,
+        label = accountInfo.label.map(AccountLabel(_))
```
Comment (vbergeron-ledger) [inline account-manager/src/main/scala/co/ledger/lama/manager/Service.scala:194]:
```suggestion
        label = accountInfo.label.map(AccountLabel)
```

File: account-manager/src/main/scala/co/ledger/lama/manager/Service.scala [hunk 3]
```diff
@@ -216,20 +219,21 @@ class Service(val db: Transactor[IO], val coinConfigs: List[CoinConfig])
...
+            AccountInfoResult(
+              UuidUtils.uuidToBytes(account.id),
+              account.key,
+              account.syncFrequency.toSeconds,
+              Some(
+                Se(
+                  UuidUtils.uuidToBytes(account.syncId),
+                  account.status.name,
+                  ByteString.copyFrom(account.payload.asJson.noSpaces.getBytes())
+                )
```
Comment (vbergeron-ledger) [inline account-manager/src/main/scala/co/ledger/lama/manager/Service.scala:233]: Please extract it to a relevant method in either `SyncEvent` object or `Account` class.
Also `Se` is not very clear.

**Review by vbergeron-ledger — CHANGES_REQUESTED** [review]

File: account-manager/src/main/scala/co/ledger/lama/manager/Service.scala (hunk 3 above)
Comment (mbasire-ledger) [reply account-manager/src/main/scala/co/ledger/lama/manager/Service.scala:233]: Ok, i will extract it but it is out of scope of my current issue.

**Review by nukky94 — CHANGES_REQUESTED** [review]
Could you please also add the ability to update the label into `PUT /accounts/{id}` endpoint?

File: account-manager/src/main/scala/co/ledger/lama/manager/Service.scala (hunk 3 above)
Comment (mbasire-ledger) [reply account-manager/src/main/scala/co/ledger/lama/manager/Service.scala:233]: Se seems to be an alias of SyncEvent, i do not known what is the purpose of this alias.

File: account-manager/src/main/scala/co/ledger/lama/manager/Service.scala [hunk 4]
```diff
@@ -16,7 +13,7 @@ import co.ledger.lama.manager.Exceptions.{
 }
 import co.ledger.lama.manager.config.CoinConfig
 import co.ledger.lama.manager.models.AccountInfo
-import co.ledger.lama.manager.protobuf.{SyncEvent => Se, _}
+import co.ledger.lama.manager.protobuf._
```
Comment (nukky94) [inline account-manager/src/main/scala/co/ledger/lama/manager/Service.scala:16]: Actually it's better to import:
`import co.ledger.lama.common.models._`
and `import co.ledger.lama.manager.protobuf`

Then use pb models prefixed by `protobuf`

Comment (mbasire-ledger) [reply account-manager/src/main/scala/co/ledger/lama/manager/Service.scala:16]: There is more protobuf references in Service.scala than models references. Service implements AccountManagerServiceFs2Grpc which depends exclusively on protobuf references.

**Review by nukky94 — APPROVED** [review]

File: account-manager/src/main/scala/co/ledger/lama/manager/Service.scala (hunk 2 above)
Comment (mbasire-ledger) [reply account-manager/src/main/scala/co/ledger/lama/manager/Service.scala:195]: This change won't compile. AccountLabel.apply takes 2 parameters.

**Review by vbergeron-ledger — APPROVED** [review]

**Pull request merged**